### PR TITLE
Remove Solrizer dependency

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

**Comment by no-reply** [comment]
I agree with @cjcolvar that the namespace is an issue. I'd propose to alias `Solrizer::InvalidIndexDescriptor` to `InvalidIndexDescriptor` so that type checks continue to work.

Also, `InvalidIndexDescriptor` currently inherits `Solrizer::SolrizerError` (and `RuntimeError`). I think we need to define that class and insert it into the inheritance tree as well.

File: lib/active_fedora/indexing/descriptor.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def initialize(*args)
           @type_required = opts[:requires_type]
         end
         @index_type = args
-        raise Solrizer::InvalidIndexDescriptor, "Invalid index type passed to Sorizer.solr_name.  It should be an array like [:string, :indexed, :stored, :multivalued]. You provided: `#{@index_type}'" unless index_type.is_a? Array
+        raise InvalidIndexDescriptor, "Invalid index type passed to Sorizer.solr_name.  It should be an array like [:string, :indexed, :stored, :multivalued]. You provided: `#{@index_type}'" unless index_type.is_a? Array
```
Comment (jrgriffiniii) [inline lib/active_fedora/indexing/descriptor.rb]: Properly ports https://github.com/samvera/solrizer/blob/master/lib/solrizer/descriptor.rb#L11

File: lib/active_fedora/indexing/field_mapper.rb [hunk 2]
```diff
@@ -87,7 +87,7 @@ def solr_names_and_values(field_name, field_value, index_types)
               values = (results[name] ||= [])
               values << value unless value.nil? || values.include?(value)
             else
-              Solrizer.logger.warn "Setting #{name} to `#{value}', but it already had `#{results[name]}'" if results[name] && Solrizer.logger
+              logger.warn "Setting #{name} to `#{value}', but it already had `#{results[name]}'" if results[name] && logger
```
Comment (jrgriffiniii) [inline lib/active_fedora/indexing/field_mapper.rb]: Properly ports https://github.com/samvera/solrizer/blob/master/lib/solrizer/field_mapper.rb#L228

File: lib/active_fedora/indexing/field_mapper.rb [hunk 3]
```diff
@@ -109,7 +109,7 @@ def indexer(index_type)
                        when Descriptor
                          index_type
                        else
-                         raise Solrizer::InvalidIndexDescriptor, "#{index_type.class} is not a valid indexer_type. Use a String, Symbol or Descriptor."
+                         raise InvalidIndexDescriptor, "#{index_type.class} is not a valid indexer_type. Use a String, Symbol or Descriptor."
```
Comment (jrgriffiniii) [inline lib/active_fedora/indexing/field_mapper.rb]: Properly ports https://github.com/samvera/solrizer/blob/master/lib/solrizer/field_mapper.rb#L155

File: lib/active_fedora/indexing/suffix.rb [hunk 4]
```diff
@@ -28,7 +28,7 @@ def data_type
       end
 
       def to_s
-        raise Solrizer::InvalidIndexDescriptor, "Missing datatype for #{@fields}" unless data_type
+        raise InvalidIndexDescriptor, "Missing datatype for #{@fields}" unless data_type
```
Comment (jrgriffiniii) [inline lib/active_fedora/indexing/suffix.rb]: https://github.com/samvera/solrizer/blob/master/lib/solrizer/suffix.rb#L32 👍

File: lib/active_fedora/indexing/suffix.rb [hunk 5]
```diff
@@ -65,7 +65,7 @@ def self.config
                                                    when :long
                                                      'lt'
                                                    else
-                                                     raise Solrizer::InvalidIndexDescriptor, "Invalid datatype `#{type.inspect}'. Must be one of: :date, :time, :text, :text_en, :string, :symbol, :integer, :boolean"
+                                                     raise InvalidIndexDescriptor, "Invalid datatype `#{type.inspect}'. Must be one of: :date, :time, :text, :text_en, :string, :symbol, :integer, :boolean"
```
Comment (jrgriffiniii) [inline lib/active_fedora/indexing/suffix.rb]: https://github.com/samvera/solrizer/blob/master/lib/solrizer/suffix.rb#L69 👍

**Comment by jrgriffiniii** [comment]
This was deprecated by https://github.com/samvera/active_fedora/pull/1384

**Pull request closed without merging**